ddl/ddl_worker.go

```diff
@@ -108,6 +113,30 @@ func (w *worker) close() {
 	log.Infof("[ddl-%s] close DDL worker", w)
 }
 
+// getSessionCtx get sessionctx from context resource pool.
+// Please remember to call putSessionCtx after you getting the sessionctx.
+func (w *worker) getSessionCtx() (sessionctx.Context, error) {
+	if w.ctxPool == nil {
+		return mock.NewContext(), nil
+	}
+	resource, err := w.ctxPool.Get()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	ctx := resource.(sessionctx.Context)
+	ctx.GetSessionVars().SetStatusFlag(mysql.ServerStatusAutocommit, true)
+	ctx.GetSessionVars().InRestrictedSQL = true
+	return ctx, nil
+}
+
+// putSessionCtx return sessionctx to context resource pool.
+func (w *worker) putSessionCtx(ctx sessionctx.Context) {
+	if w.ctxPool != nil {
+		w.ctxPool.Put(ctx.(pools.Resource))
```

Comment:
`w.ctxPool.Put(ctx)` works the same?

Reply:
Should we reset the `Autocommit` and `InRestrictedSQL`?

Reply:
@lamxTyler No needs, the session is only using to run  restricted-sql. Actually, in delRangeManager, we don't reset these fields before we put it into ctxPool.

Reply:
@tiancaiamao sessionctx.Context don't implement Close method, but *session does.

Reply:
Oh, I see.